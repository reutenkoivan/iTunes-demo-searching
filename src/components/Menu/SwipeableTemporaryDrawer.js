import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer';
import { Link as RouterLink } from 'react-router-dom'
import Link from '@material-ui/core/Link';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Favorite from "@material-ui/icons/Favorite"
import HomeIcon from "@material-ui/icons/Home"

const styles = {
  list: {
    width: 250,
  }
};

class SwipeableTemporaryDrawer extends React.Component {

  toggleDrawer = open => this.props.toggleDrawer(open);

  render() {
    const { classes } = this.props;

    return (
      <div>
        <SwipeableDrawer
          open={this.props.isOpen}
          onClose={this.toggleDrawer(false)}
          onOpen={this.toggleDrawer(true)}
        >
          <div
            role="button"
            onClick={this.toggleDrawer(false)}
            onKeyDown={this.toggleDrawer(false)}
          >
            <div className={classes.list}>
              <List>
                <Link component={RouterLink} to="/">
                  <ListItem button >
                    <HomeIcon/>
                    <ListItemText primary={"Home"} />
                  </ListItem>
                </Link>
                <Link component={RouterLink} to="/favorites">
                  <ListItem button >
                    <Favorite />
                    <ListItemText primary={"Favorites"} />
                  </ListItem>
                </Link>
              </List>
              <Divider />
            </div>
          </div>
        </SwipeableDrawer>
      </div>
    );
  }
}

export default withStyles(styles)(SwipeableTemporaryDrawer);
