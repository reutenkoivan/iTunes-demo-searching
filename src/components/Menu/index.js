import React, { Component } from "react";
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import InputBase from '@material-ui/core/InputBase';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { withStyles } from '@material-ui/core/styles';
import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';
import SwipeableTemporaryDrawer from "../Menu/SwipeableTemporaryDrawer"


const styles = theme => ({
  root: {
    width: '100%'
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  title: {
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25),
    },
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit,
      width: 'auto',
    },
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputRoot: {
    color: 'inherit',
    width: '100%',
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: 120,
      '&:focus': {
        width: 250,
      },
    },
  },
});

class Menu extends Component {
  state = {
    isOpen: false,
    inputValue: ""
  };

  searchingValue = e => {
    const linkParams = e.target.value.trim().split(/\s+/).join("+")
    this.props.getSongs(linkParams)
    this.setState({inputValue: linkParams})
  }

  toggleDrawer = open => () => {
    this.setState({
      isOpen: open,
    });
  };

  render() {
    const { classes } = this.props;

    return (
      <React.Fragment>
        <div className={classes.root}>
          <AppBar position="static">
            <Toolbar>
              <IconButton onClick={this.toggleDrawer(true)} className={classes.menuButton} color="inherit" aria-label="Open drawer">
                <MenuIcon />
              </IconButton>
              <Typography className={classes.title} variant="h6" color="inherit" noWrap>
                Searching songs
              </Typography>
              <div className={classes.grow} />
              <div className={classes.search}>
                <div className={classes.searchIcon}>
                  <SearchIcon />
                </div>
                <InputBase
                  onChange={this.searchingValue}
                  placeholder="Songs..."
                  classes={{
                    root: classes.inputRoot,
                    input: classes.inputInput,
                  }}
                />
              </div>
            </Toolbar>
          </AppBar>
        </div>
        <SwipeableTemporaryDrawer isOpen={this.state.isOpen} toggleDrawer={this.toggleDrawer}/>
      </React.Fragment>
    );
  }
}

export default withStyles(styles)(Menu);