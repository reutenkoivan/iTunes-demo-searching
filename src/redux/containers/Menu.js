import { connect } from "react-redux";
import Menu from "../../components/Menu";
import fetchSongs from "../actionCreators/fetchSongs";

const mapDispatchToProps = dispatch => ({
  getSongs: (str) => dispatch(fetchSongs(str))
});

export default connect(null, mapDispatchToProps)(Menu);
