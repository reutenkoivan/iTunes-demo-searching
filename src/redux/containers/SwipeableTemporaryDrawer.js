import { connect } from "react-redux";
import SwipeableTemporaryDrawer from "../../components/Menu/SwipeableTemporaryDrawer"
import addToLocalStorage from "../actionCreators/addToLocalStorage";

const mapStateToProps = state => ({
  favorites: state.favorites,
  rootPath: state.rootPath
});


const mapDispatchToProps = dispatch => ({
  switchFavoritesState: (song) => dispatch(addToLocalStorage(song))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SwipeableTemporaryDrawer);
