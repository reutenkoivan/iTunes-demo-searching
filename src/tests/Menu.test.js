import React from 'react';
import Menu from "../components/Menu";
import SwipeableTemporaryDrawer from "../redux/containers/SwipeableTemporaryDrawer"
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createShallow, getClasses } from '@material-ui/core/test-utils';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer';

configure({ adapter: new Adapter() });
const song = {
  artistName: "artistName",
  trackName: "trackName",
  previewUrl: "previewUrl",
  artworkUrl: "artworkUrl100",
  trackPrice: "trackPrice",
  trackId: "trackId",
  collectionName: "collectionName",
  artistViewUrl: "artistViewUrl",
  isFavorite: false
};
const playerInfo = {
  trackLink: "previewUrl",
  trackId: "trackId",
  artistName: "artistName",
  trackName: "trackName",
  playlist: [song, song],
  index: 1,
  next: 2,
  isPlaying: false,
  previous: 0
};
const history = {location: {pathname: ""}};

let shallow;
let classes;

beforeAll(() => {
  shallow = createShallow({ dive: true });
  classes = getClasses(<Menu history={history}/>);
});

it('renders without crashing', () => {
  const wrap = shallow(<Menu history={history}/>);
  expect(wrap).toMatchSnapshot()
});

it('change state', () => {
  const wrap = shallow(<Menu history={history}/>);
  expect(wrap.state().isOpen).toEqual(false);
  wrap.find(IconButton).simulate("click");
  expect(wrap.state().isOpen).toEqual(true);
});

it('is correct title for SearchingPage', () => {
  const wrap = shallow(<Menu history={history}/>);
  expect(wrap.find(Typography).dive().dive().text()).toEqual("Searching page")
});

it('is correct title for FavoritePage', () => {
  const favoriteHistory = {location: {pathname: "/favorites"}};
  const wrap = shallow(<Menu history={favoriteHistory}/>);
  expect(wrap.find(Typography).dive().dive().text()).toEqual("Favorite songs")
});

it('doesnt showing searching field for FavoritePage', () => {
  const favoriteHistory = {location: {pathname: "/favorites"}};
  const wrap = shallow(<Menu history={favoriteHistory}/>);
  expect(wrap.find(`.${classes.search}`)).toHaveLength(0);
});

it('showing searching field for SearchingPage', () => {
  const wrap = shallow(<Menu history={history}/>);
  expect(wrap.find(`.${classes.search}`)).toHaveLength(1);
});
