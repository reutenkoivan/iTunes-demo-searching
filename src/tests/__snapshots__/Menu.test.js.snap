// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    classes={
      Object {
        "grow": "Menu-grow-2",
        "inputInput": "Menu-inputInput-8",
        "inputRoot": "Menu-inputRoot-7",
        "menuButton": "Menu-menuButton-3",
        "root": "Menu-root-1",
        "search": "Menu-search-5",
        "searchIcon": "Menu-searchIcon-6",
        "title": "Menu-title-4",
      }
    }
    history={
      Object {
        "location": Object {
          "pathname": "",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="Menu-root-1"
        >
          <WithStyles(AppBar)
            position="static"
          >
            <WithStyles(Toolbar)>
              <WithStyles(IconButton)
                aria-label="Open drawer"
                className="Menu-menuButton-3"
                color="inherit"
                onClick={[Function]}
              >
                <pure(MenuIcon) />
              </WithStyles(IconButton)>
              <WithStyles(Typography)
                className="Menu-title-4"
                color="inherit"
                noWrap={true}
                variant="h6"
              >
                Searching page
              </WithStyles(Typography)>
              <div
                className="Menu-grow-2"
              />
              <div
                className="Menu-search-5"
              >
                <div
                  className="Menu-searchIcon-6"
                >
                  <pure(SearchIcon) />
                </div>
                <DebounceInput
                  classes={
                    Object {
                      "input": "Menu-inputInput-8",
                      "root": "Menu-inputRoot-7",
                    }
                  }
                  debounceTimeout={700}
                  element={[Function]}
                  forceNotifyByEnter={true}
                  forceNotifyOnBlur={true}
                  minLength={2}
                  onChange={[Function]}
                  placeholder="Song..."
                  type="text"
                />
              </div>
            </WithStyles(Toolbar)>
          </WithStyles(AppBar)>
        </div>,
        <Connect(WithStyles(SwipeableTemporaryDrawer))
          isOpen={false}
          toggleDrawer={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(AppBar)
            position="static"
          >
            <WithStyles(Toolbar)>
              <WithStyles(IconButton)
                aria-label="Open drawer"
                className="Menu-menuButton-3"
                color="inherit"
                onClick={[Function]}
              >
                <pure(MenuIcon) />
              </WithStyles(IconButton)>
              <WithStyles(Typography)
                className="Menu-title-4"
                color="inherit"
                noWrap={true}
                variant="h6"
              >
                Searching page
              </WithStyles(Typography)>
              <div
                className="Menu-grow-2"
              />
              <div
                className="Menu-search-5"
              >
                <div
                  className="Menu-searchIcon-6"
                >
                  <pure(SearchIcon) />
                </div>
                <DebounceInput
                  classes={
                    Object {
                      "input": "Menu-inputInput-8",
                      "root": "Menu-inputRoot-7",
                    }
                  }
                  debounceTimeout={700}
                  element={[Function]}
                  forceNotifyByEnter={true}
                  forceNotifyOnBlur={true}
                  minLength={2}
                  onChange={[Function]}
                  placeholder="Song..."
                  type="text"
                />
              </div>
            </WithStyles(Toolbar)>
          </WithStyles(AppBar)>,
          "className": "Menu-root-1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Toolbar)>
              <WithStyles(IconButton)
                aria-label="Open drawer"
                className="Menu-menuButton-3"
                color="inherit"
                onClick={[Function]}
              >
                <pure(MenuIcon) />
              </WithStyles(IconButton)>
              <WithStyles(Typography)
                className="Menu-title-4"
                color="inherit"
                noWrap={true}
                variant="h6"
              >
                Searching page
              </WithStyles(Typography)>
              <div
                className="Menu-grow-2"
              />
              <div
                className="Menu-search-5"
              >
                <div
                  className="Menu-searchIcon-6"
                >
                  <pure(SearchIcon) />
                </div>
                <DebounceInput
                  classes={
                    Object {
                      "input": "Menu-inputInput-8",
                      "root": "Menu-inputRoot-7",
                    }
                  }
                  debounceTimeout={700}
                  element={[Function]}
                  forceNotifyByEnter={true}
                  forceNotifyOnBlur={true}
                  minLength={2}
                  onChange={[Function]}
                  placeholder="Song..."
                  type="text"
                />
              </div>
            </WithStyles(Toolbar)>,
            "position": "static",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(IconButton)
                  aria-label="Open drawer"
                  className="Menu-menuButton-3"
                  color="inherit"
                  onClick={[Function]}
                >
                  <pure(MenuIcon) />
                </WithStyles(IconButton)>,
                <WithStyles(Typography)
                  className="Menu-title-4"
                  color="inherit"
                  noWrap={true}
                  variant="h6"
                >
                  Searching page
                </WithStyles(Typography)>,
                <div
                  className="Menu-grow-2"
                />,
                <div
                  className="Menu-search-5"
                >
                  <div
                    className="Menu-searchIcon-6"
                  >
                    <pure(SearchIcon) />
                  </div>
                  <DebounceInput
                    classes={
                      Object {
                        "input": "Menu-inputInput-8",
                        "root": "Menu-inputRoot-7",
                      }
                    }
                    debounceTimeout={700}
                    element={[Function]}
                    forceNotifyByEnter={true}
                    forceNotifyOnBlur={true}
                    minLength={2}
                    onChange={[Function]}
                    placeholder="Song..."
                    type="text"
                  />
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-label": "Open drawer",
                  "children": <pure(MenuIcon) />,
                  "className": "Menu-menuButton-3",
                  "color": "inherit",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Searching page",
                  "className": "Menu-title-4",
                  "color": "inherit",
                  "noWrap": true,
                  "variant": "h6",
                },
                "ref": null,
                "rendered": "Searching page",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "Menu-grow-2",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="Menu-searchIcon-6"
                    >
                      <pure(SearchIcon) />
                    </div>,
                    <DebounceInput
                      classes={
                        Object {
                          "input": "Menu-inputInput-8",
                          "root": "Menu-inputRoot-7",
                        }
                      }
                      debounceTimeout={700}
                      element={[Function]}
                      forceNotifyByEnter={true}
                      forceNotifyOnBlur={true}
                      minLength={2}
                      onChange={[Function]}
                      placeholder="Song..."
                      type="text"
                    />,
                  ],
                  "className": "Menu-search-5",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <pure(SearchIcon) />,
                      "className": "Menu-searchIcon-6",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "classes": Object {
                        "input": "Menu-inputInput-8",
                        "root": "Menu-inputRoot-7",
                      },
                      "debounceTimeout": 700,
                      "element": [Function],
                      "forceNotifyByEnter": true,
                      "forceNotifyOnBlur": true,
                      "inputRef": undefined,
                      "minLength": 2,
                      "onBlur": undefined,
                      "onChange": [Function],
                      "onKeyDown": undefined,
                      "placeholder": "Song...",
                      "type": "text",
                      "value": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isOpen": false,
          "toggleDrawer": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="Menu-root-1"
          >
            <WithStyles(AppBar)
              position="static"
            >
              <WithStyles(Toolbar)>
                <WithStyles(IconButton)
                  aria-label="Open drawer"
                  className="Menu-menuButton-3"
                  color="inherit"
                  onClick={[Function]}
                >
                  <pure(MenuIcon) />
                </WithStyles(IconButton)>
                <WithStyles(Typography)
                  className="Menu-title-4"
                  color="inherit"
                  noWrap={true}
                  variant="h6"
                >
                  Searching page
                </WithStyles(Typography)>
                <div
                  className="Menu-grow-2"
                />
                <div
                  className="Menu-search-5"
                >
                  <div
                    className="Menu-searchIcon-6"
                  >
                    <pure(SearchIcon) />
                  </div>
                  <DebounceInput
                    classes={
                      Object {
                        "input": "Menu-inputInput-8",
                        "root": "Menu-inputRoot-7",
                      }
                    }
                    debounceTimeout={700}
                    element={[Function]}
                    forceNotifyByEnter={true}
                    forceNotifyOnBlur={true}
                    minLength={2}
                    onChange={[Function]}
                    placeholder="Song..."
                    type="text"
                  />
                </div>
              </WithStyles(Toolbar)>
            </WithStyles(AppBar)>
          </div>,
          <Connect(WithStyles(SwipeableTemporaryDrawer))
            isOpen={false}
            toggleDrawer={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(AppBar)
              position="static"
            >
              <WithStyles(Toolbar)>
                <WithStyles(IconButton)
                  aria-label="Open drawer"
                  className="Menu-menuButton-3"
                  color="inherit"
                  onClick={[Function]}
                >
                  <pure(MenuIcon) />
                </WithStyles(IconButton)>
                <WithStyles(Typography)
                  className="Menu-title-4"
                  color="inherit"
                  noWrap={true}
                  variant="h6"
                >
                  Searching page
                </WithStyles(Typography)>
                <div
                  className="Menu-grow-2"
                />
                <div
                  className="Menu-search-5"
                >
                  <div
                    className="Menu-searchIcon-6"
                  >
                    <pure(SearchIcon) />
                  </div>
                  <DebounceInput
                    classes={
                      Object {
                        "input": "Menu-inputInput-8",
                        "root": "Menu-inputRoot-7",
                      }
                    }
                    debounceTimeout={700}
                    element={[Function]}
                    forceNotifyByEnter={true}
                    forceNotifyOnBlur={true}
                    minLength={2}
                    onChange={[Function]}
                    placeholder="Song..."
                    type="text"
                  />
                </div>
              </WithStyles(Toolbar)>
            </WithStyles(AppBar)>,
            "className": "Menu-root-1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Toolbar)>
                <WithStyles(IconButton)
                  aria-label="Open drawer"
                  className="Menu-menuButton-3"
                  color="inherit"
                  onClick={[Function]}
                >
                  <pure(MenuIcon) />
                </WithStyles(IconButton)>
                <WithStyles(Typography)
                  className="Menu-title-4"
                  color="inherit"
                  noWrap={true}
                  variant="h6"
                >
                  Searching page
                </WithStyles(Typography)>
                <div
                  className="Menu-grow-2"
                />
                <div
                  className="Menu-search-5"
                >
                  <div
                    className="Menu-searchIcon-6"
                  >
                    <pure(SearchIcon) />
                  </div>
                  <DebounceInput
                    classes={
                      Object {
                        "input": "Menu-inputInput-8",
                        "root": "Menu-inputRoot-7",
                      }
                    }
                    debounceTimeout={700}
                    element={[Function]}
                    forceNotifyByEnter={true}
                    forceNotifyOnBlur={true}
                    minLength={2}
                    onChange={[Function]}
                    placeholder="Song..."
                    type="text"
                  />
                </div>
              </WithStyles(Toolbar)>,
              "position": "static",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(IconButton)
                    aria-label="Open drawer"
                    className="Menu-menuButton-3"
                    color="inherit"
                    onClick={[Function]}
                  >
                    <pure(MenuIcon) />
                  </WithStyles(IconButton)>,
                  <WithStyles(Typography)
                    className="Menu-title-4"
                    color="inherit"
                    noWrap={true}
                    variant="h6"
                  >
                    Searching page
                  </WithStyles(Typography)>,
                  <div
                    className="Menu-grow-2"
                  />,
                  <div
                    className="Menu-search-5"
                  >
                    <div
                      className="Menu-searchIcon-6"
                    >
                      <pure(SearchIcon) />
                    </div>
                    <DebounceInput
                      classes={
                        Object {
                          "input": "Menu-inputInput-8",
                          "root": "Menu-inputRoot-7",
                        }
                      }
                      debounceTimeout={700}
                      element={[Function]}
                      forceNotifyByEnter={true}
                      forceNotifyOnBlur={true}
                      minLength={2}
                      onChange={[Function]}
                      placeholder="Song..."
                      type="text"
                    />
                  </div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "aria-label": "Open drawer",
                    "children": <pure(MenuIcon) />,
                    "className": "Menu-menuButton-3",
                    "color": "inherit",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Searching page",
                    "className": "Menu-title-4",
                    "color": "inherit",
                    "noWrap": true,
                    "variant": "h6",
                  },
                  "ref": null,
                  "rendered": "Searching page",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "Menu-grow-2",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="Menu-searchIcon-6"
                      >
                        <pure(SearchIcon) />
                      </div>,
                      <DebounceInput
                        classes={
                          Object {
                            "input": "Menu-inputInput-8",
                            "root": "Menu-inputRoot-7",
                          }
                        }
                        debounceTimeout={700}
                        element={[Function]}
                        forceNotifyByEnter={true}
                        forceNotifyOnBlur={true}
                        minLength={2}
                        onChange={[Function]}
                        placeholder="Song..."
                        type="text"
                      />,
                    ],
                    "className": "Menu-search-5",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <pure(SearchIcon) />,
                        "className": "Menu-searchIcon-6",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "classes": Object {
                          "input": "Menu-inputInput-8",
                          "root": "Menu-inputRoot-7",
                        },
                        "debounceTimeout": 700,
                        "element": [Function],
                        "forceNotifyByEnter": true,
                        "forceNotifyOnBlur": true,
                        "inputRef": undefined,
                        "minLength": 2,
                        "onBlur": undefined,
                        "onChange": [Function],
                        "onKeyDown": undefined,
                        "placeholder": "Song...",
                        "type": "text",
                        "value": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isOpen": false,
            "toggleDrawer": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
  Symbol(enzyme.__childContext__): null,
}
`;
